pkg/sync/v6/sync.go

```diff
@@ -302,6 +310,20 @@ var (
 // objects with db.
 // TODO: need to implement deleting objects which we don't want any more.
 func (s *sync) writeDB() error {
+	// openshift namespace is special. It hosts sharedResources.
+	// Reconcile protection logic is reconcileProtected func,
+	// but we check if openshift namespace has protection annotation
+	// to check if we need to apply sharedResourceFilter or not
+	// default - resources are managed
+	managedSharedResources := true
+	namespace, err := s.kc.CoreV1().Namespaces().Get("openshift", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if strings.ToLower(namespace.GetAnnotations()[reconcileProtectAnnotationKey]) == "true" {
```

Comment:
are you missing a `namespace != nil &&` here?

Reply:
`openshift` namespace exists with the start of static pods. But we can add something like that just to be sure